addons/dialogue_manager/constants.gd

```diff
@@ -216,11 +220,11 @@ static func get_error_message(error: int) -> String:
 
 
 static func translate(string: String) -> String:
-	var base_path = new().get_script().resource_path.get_base_dir()
-
-	var language: String = TranslationServer.get_tool_locale()
-	var translations_path: String = "%s/l10n/%s.po" % [base_path, language]
-	var fallback_translations_path: String = "%s/l10n/%s.po" % [base_path, TranslationServer.get_tool_locale().substr(0, 2)]
-	var en_translations_path: String = "%s/l10n/en.po" % base_path
-	var translations: Translation = load(translations_path if FileAccess.file_exists(translations_path) else (fallback_translations_path if FileAccess.file_exists(fallback_translations_path) else en_translations_path))
-	return translations.get_message(string)
+	var locale: String = TranslationServer.get_tool_locale()
+	if _current_locale != locale:
+		var base_path: String = new().get_script().resource_path.get_base_dir()
+		var translation_path: String = "%s/l10n/%s.po" % [base_path, locale]
+		var fallback_translation_path: String = "%s/l10n/%s.po" % [base_path, locale.substr(0, 2)]
+		var en_translation_path: String = "%s/l10n/en.po" % base_path
+		_current_translation = load(translation_path if FileAccess.file_exists(translation_path) else (fallback_translation_path if FileAccess.file_exists(fallback_translation_path) else en_translation_path))
```

Comment:
Should this be setting the _current_locale? Otherwise wouldn't we still be fetching every time?

Reply:
Good catch.